### PR TITLE
Add unit tests for upgrade tool

### PR DESCRIPTION
**Pull request by christiancp100**
### What does it do?

Adds tests for the upgrade tool

### Why is it needed?

To test the upgrade tool

### CODE REVIEW

**Review by innerdvations — APPROVED** [review]
LGTM!

**Comment by innerdvations** [comment]
...except there are some failing tests, relating to 'fs', could be related to the same thing @Convly just fixed in another branch?

**Comment by Convly** [comment]
> ...except there are some failing tests, relating to 'fs', could be related to the same thing @Convly just fixed in another branch?

Yeah, you may want to wait for #19053 to be merged?

File: packages/utils/upgrade/src/modules/codemod-repository/__tests__/repository.test.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
+import { vol, fs } from 'memfs';
+
+import { Range, SemVer } from 'semver';
+import {
```
Comment (Convly) [inline packages/utils/upgrade/src/modules/codemod-repository/__tests__/repository.test.ts:4]: I have a doubt all of a sudden, but aren't we supposed to import those after mocking fs?

Comment (christiancp100) [reply packages/utils/upgrade/src/modules/codemod-repository/__tests__/repository.test.ts:4]: why? 😅

**Review by innerdvations — APPROVED** [review]
LGTM! Tests are passing now too.

**Pull request merged**